File: ruby/matrix/matrix.rb
=begin
Write your code for the 'Matrix' exercise in this file. Make the tests in
`matrix_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/matrix` directory.
=end
class Matrix

    attr_accessor :rows
    attr_accessor :columns

    def initialize(matrix)
        # parse and save the matrix to class variable value
        # iteration 2 changes 
        # - use transpose it is a built-in-method to do what I want for columns
        # - I should use a &:to_i with map instead of a loop, gotta get used to ruby-isms.
        # iteration 3 changes
        # - Comments admitting to changes swiped from the community solutions.
        @rows = []
        @columns = [] 
        matrix.each_line {|row|
            @rows.push(row.chomp.split.map(&:to_i))
        }
        @columns = @rows.transpose
    end
end

File: javascript/matrix/matrix.js
//
// This is only a SKELETON file for the 'Matrix' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Matrix {
  constructor(data) {
    this.matrix = data.split('\n').map((row) => row.split(' ').map((item) => Number(item)))
    let colCount = Math.max(... this.matrix.map((row) => row.length))
    this.transpose = Array.from({length: colCount},(_, i) => i).map((n) => this.matrix.map((row) => row[n]));
  }

  get rows() {
    return this.matrix
  }

  get columns() {
    return this.transpose
  }
}
